packages/next-auth/src/providers/battlenet.ts

```diff
@@ -0,0 +1,32 @@
+import type { OAuthConfig } from "."
+import OAuthProvider from './auth0'
+export interface BattleNetProfile {
+  sub: string
+  battle_tag: string
+}
+
+export default function BattleNet(
+  options: { clientId: string, clientSecret: string, region: 'US' | 'EU' | 'KR' | 'TW' | 'CN' }
+): OAuthConfig<BattleNetProfile> {
```

Comment:
```suggestion
export default function BattleNet<
  P extends Record<string, any> = BattleNetProfile
>(
  options: OAuthUserConfig<P> & {
    /** Defaults to `"CN"` */
    region?: "US" | "EU" | "KR" | "TW" | "CN"
  }
): OAuthConfig<P> {
```

Reply:
I assumed that we wouldn't want to allow people to override the "working" config :D